src/openms/include/OpenMS/FORMAT/MzTabMFile.h

```diff
@@ -0,0 +1,78 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2020.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Oliver Alka $
+// $Authors: Oliver Alka  $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/FORMAT/MzTabM.h>
+
+namespace OpenMS
+{
+  class String;
+  class SVOutStream;
+
+  /**
+    @brief File adapter for MzTab-M files
+
+    @ingroup FileIO
+  */
+
+  class OPENMS_DLLAPI MzTabMFile
+  {
+  public:
+    /// Default Constructor
+    MzTabMFile();
+
+    /// Default Destructor
+    ~MzTabMFile();
+
+    /// Store MzTabM file
+    void store(const String& filename, const MzTabM& mztab_m) const;
+
+  protected:
+    void generateMzTabMMetaDataSection_(const MzTabMMetaData& map, StringList& sl) const;
+
+    String generateMzTabMSmallMoleculeHeader_(const MzTabMMetaData& meta, const std::vector<String>& optional_columns, size_t& n_columns) const;
```

Comment:
why does n_columns need to be passed? please add doc

Reply:
Add documentation. FYI: This stores the number of columns used in the header or the row, which is later used in a postcondition, to sanity check if the header and rows have the same length (columnwise).